container.go

```diff
@@ -63,24 +63,23 @@ func (c *Container) Provide(t interface{}) error {
 	defer c.Unlock()
 
 	ctype := reflect.TypeOf(t)
-
 	switch ctype.Kind() {
 	case reflect.Func:
-		switch ctype.NumOut() {
-		case 2:
-			if ctype.Out(1) != _typeOfError {
-				return errReturnErrKind
-			}
-			fallthrough
-		case 1:
+		count := ctype.NumOut()
+		if count == 0 {
+			return errReturnCount
+		} else if count == 1 {
 			objType := ctype.Out(0)
 			if objType.Kind() != reflect.Ptr && objType.Kind() != reflect.Interface {
 				return errReturnKind
 			}
-		default:
-			return errReturnCount
+		} else {
+			// if last variable is an error, we will not add it to the graph
+			if ctype.Out(count-1) == _typeOfError {
+				count--
```

Comment:
i'm not sure why this count var is here, since we've already done the error checking. Can't this go inside the `registerConstructor` handling so we have one less variable to pass?